client/astarte_interface.go

```diff
@@ -0,0 +1,285 @@
+type AstarteInterfaceAggregation int
+
+const (
+	// IndividualAggregation represents an interface with individual endpoints
+	IndividualAggregation AstarteInterfaceAggregation = iota
+	// ObjectAggregation represents an interface with aggregated endpoints
+	ObjectAggregation
+)
+
+func (s AstarteInterfaceAggregation) String() string {
+	return astarteInterfaceAggregationToString[s]
+}
+
+var astarteInterfaceAggregationToString = map[AstarteInterfaceAggregation]string{
+	IndividualAggregation: "individual",
+	ObjectAggregation:     "object",
+}
+
+var astarteInterfaceAggregationToID = map[string]AstarteInterfaceAggregation{
+	"individual": IndividualAggregation,
+	"object":     ObjectAggregation,
+}
+
+// MarshalJSON marshals the enum as a quoted json string
+func (s AstarteInterfaceAggregation) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(astarteInterfaceAggregationToString[s])
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
+
+// UnmarshalJSON unmashals a quoted json string to the enum value
+func (s *AstarteInterfaceAggregation) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+	// Note that if the string cannot be found then it will be set to the zero value, 'IndividualAggregation' in this case.
+	*s = astarteInterfaceAggregationToID[j]
+	return nil
+}
+
+// AstarteMappingReliability represents the reliability of a mapping
+type AstarteMappingReliability int
+
+const (
+	// UniqueReliability represents a QoS 2-like reliability on the wire
+	UniqueReliability AstarteMappingReliability = iota
+	// GuaranteedReliability represents a QoS 1-like reliability on the wire
+	GuaranteedReliability
+	// UnreliableReliability represents a QoS 0-like reliability on the wire
+	UnreliableReliability
+)
+
+func (s AstarteMappingReliability) String() string {
+	return astarteMappingReliabilityToString[s]
+}
+
+var astarteMappingReliabilityToString = map[AstarteMappingReliability]string{
+	UniqueReliability:     "unique",
+	GuaranteedReliability: "guaranteed",
+	UnreliableReliability: "unreliable",
+}
+
+var astarteMappingReliabilityToID = map[string]AstarteMappingReliability{
+	"unique":     UniqueReliability,
+	"guaranteed": GuaranteedReliability,
+	"unreliable": UnreliableReliability,
+}
+
+// MarshalJSON marshals the enum as a quoted json string
+func (s AstarteMappingReliability) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(astarteMappingReliabilityToString[s])
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
+
+// UnmarshalJSON unmashals a quoted json string to the enum value
+func (s *AstarteMappingReliability) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+	// Note that if the string cannot be found then it will be set to the zero value, 'IndividualAggregation' in this case.
```

Comment:
Fix comment (it's not `IndividualAggregation`)
Moreover, the default value should be `UnreliablieReliability`, while here it seems that it is `UniqueReliability`